src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs

```diff
@@ -63,8 +64,8 @@ public static bool HasArgument(this IResolveFieldContext context, string name)
         {
             var isIntrospection = context.ParentType == null ? context.FieldDefinition.IsIntrospectionField() : context.ParentType.IsIntrospectionType();
             var argumentName = isIntrospection ? name : (context.Schema?.NameConverter.NameForArgument(name, context.ParentType, context.FieldDefinition) ?? name);
-            object value = null;
-            return ((context.Arguments?.TryGetValue(argumentName, out value) ?? false) && value != null);
+            ArgumentValue value = default;
+            return (context.Arguments?.TryGetValue(argumentName, out value) ?? false) && value.Source != ArgumentSource.FieldDefault;
```

Comment:
Why only `FieldDefault` and not `VariableDefault` ?

Reply:
The value was specified by the request, in the request document.  It is not a default value, as seen by the graph/execution.  How the value was supplied should not matter (literal, variable within inputs, or literal default specified for a variable).  This is why I question the validity of including `VariableDefault` at all.

Reply:
Understood.

>  This is why I question the validity of including VariableDefault at all.

I would leave this information. It may be useful to us later.